src/storage/mvcc/reader/cf_reader.rs

```diff
@@ -0,0 +1,569 @@
+        // TODO: `load_lock` should respect `fill_cache` options as well.
+        super::util::load_lock(&self.snapshot, key, &mut self.statistics)
+    }
+
+    /// Get a lock of a user key in the lock CF. If lock exists, it will be checked to
+    /// see whether it conflicts with the given `ts`. If there is no conflict or no lock,
+    /// the safe `ts` will be returned.
+    ///
+    /// Internally, there is a db `get`.
+    #[inline]
+    pub fn load_and_check_lock(&mut self, key: &Key, ts: u64) -> Result<u64> {
+        super::util::load_and_check_lock(&self.snapshot, key, ts, &mut self.statistics)
+    }
+
+    /// Iterate and get all user keys in the write CF within the given key space
+    /// (specified by `start_key` and `limit`). `limit` must not be `0`.
+    ///
+    /// Internally, there will be a `seek` operation for the first iteration and
+    /// `near_seek` for other iterations.
+    ///
+    /// The return type is `(keys, next_start_key)`. `next_start_key` is the `start_key`
+    /// that can be used to continue scanning keys. If `next_start_key` is `None`, it means
+    /// that there is no more keys.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `limit` is `0`.
+    pub fn scan_keys(
+        &mut self,
+        start_key: Option<&Key>,
+        limit: usize,
+    ) -> Result<(Vec<Key>, Option<Key>)> {
+        self.ensure_write_cursor()?;
+        let write_cursor = self.write_cursor.as_mut().unwrap();
+
+        // TODO: We need to ensure that cursor is not prefix seek.
+        assert!(limit > 0);
+
+        match start_key {
+            Some(ref x) => write_cursor.seek(x, &mut self.statistics.write)?,
+            None => write_cursor.seek_to_first(&mut self.statistics.write),
+        };
+        if !write_cursor.valid() {
+            return Ok((vec![], None));
+        }
+        let mut keys = Vec::with_capacity(limit);
+        let mut next_start_key;
+        loop {
+            let key = Key::from_encoded(write_cursor.key(&mut self.statistics.write).to_vec())
+                .truncate_ts()?;
+            // Jump to the last version of the key. We assumed that there is no key that ts == 0.
+            next_start_key = Some(key.clone().append_ts(0));
+            keys.push(key);
+            if !write_cursor.near_seek(
+                next_start_key.as_ref().unwrap(),
+                false,
+                &mut self.statistics.write,
+            )? {
+                // No more keys found, we don't need to scan keys next time
+                next_start_key = None;
+                break;
+            }
+            if keys.len() >= limit {
+                // Reach limit
+                break;
+            }
+        }
+        self.statistics.write.processed += keys.len();
+        Ok((keys, next_start_key))
+    }
+
+    /// Iterate and get all locks in the lock CF that `predicate` returns `true` within
+    /// the given key space (specified by `start_key` and `limit`). If `limit` is `0`,
```

Comment:
scan_keys' limit must larger than 0, but scan_locks' limit can be 0, this is confusing.

Reply:
Yes. This behaviour is inherited from current code. Maybe we can change it.